Wangscape/Exp.h

```diff
@@ -0,0 +1,77 @@
+#pragma once
+#include <noise/noise.h>
+#include <cmath>
+
+// This is similar to noise::module::Exp.
+// However, no rescaling is performed,
+// and no attempt is made to guard against invalid operations.
+// Make sure the range of the source module
+// is compatible with the base of the exponential!
+// This module names its parameter "base" rather than "exponent",
+// because the formula is pow(base, source_value),
+// not pow(source_value, exponent).
+class Exp : public noise::module::Module
+{
+public:
+    double base;
+    Exp() :
+        Module(GetSourceModuleCount()),
+        mBase(2.)
+    {};
+    virtual int GetSourceModuleCount() const
+    {
+        return 1;
```

Comment:
Oh, so I think setting it manually (leaving as it is) is a better solution. They could be consts.